op-challenger: Avoid loading duplicate leaves in same block

ajsutton opened the pull request:
**Description**

Fixes the large preimage input fetcher to avoid duplicating leaves when multiple transactions in the same block add leaves.

**Tests**

Updated unit tests to better reflect the block numbers recorded by contracts.
Updated the e2e tests to send larger preimages and batch send so they can be (and usually are) included in the same block. Previously the e2e tests waited for each transaction to be included in a block before sending the next.

ajsutton committed:
op-challenger: Avoid loading duplicate leaves.

mbaxter commented on op-challenger/game/keccak/fetcher/fetcher.go:
```diff
@@ -41,7 +41,16 @@ func (f *InputFetcher) FetchInputs(ctx context.Context, blockHash common.Hash, o
+	// to avoid including the leaves twice
+	lastBlock := uint64(0) // Note: can't add leaves in the genesis block as there are no transactions
 	for _, blockNum := range blockNums {
+		if blockNum == lastBlock {
```
Instead of skipping duplicates here, should we update `oracle.GetInputDataBlocks` to dedupe block numbers?  This method is used in a few places - not sure if its causing issues elsewhere.

ajsutton replied:
Good idea, that actually comes out quite nicely since we're already iterating through the results to fetch the responses anyway.

I also added an extra check that the fetched data length matches the number of bytes processed by the contract so we get a much clearer error message if the fetching is wrong which would have saved me a day or so of debugging.

mbaxter reviewed (approved):
Looks good - just a suggestion on doing the block number deduplication at a lower level

ajsutton committed:
op-challenger: Sanity check length of fetched input data.

Provides a clearer error message if the fetch is incorrect rather than just InvalidProof() revert reason.

ajsutton committed:
op-challenger: Deduplicate blocks in contract bindings.

ajsutton committed:
Deduplicate block numbers in test setup now that the contract bindings handle it.

ajsutton committed:
Fix more tests.
